engine/crates/engine-v2/src/response/write/deserialize/object/concrete.rs

```diff
@@ -218,7 +228,7 @@ impl<'de, 'ctx, 'seed> ConcreteObjectFieldsSeed<'ctx, 'seed> {
             }
         }
 
-        if response_fields.len() < self.field_shape_ids.len() {
+        if response_fields.len() < self.field_shape_ids.len() - self.skipped_ids.len() {
```

Comment:
Since we know all skipped fields will be omitted this should theoretically be fine (and not mask any field that should be there being missing)

Reply:
@Finistere please take a second look at this to make sure I'm not missing a case, thanks!

Reply:
looks sane to me, but you don't need to allocate array, just the count is enough

Reply:
Done